pkg/restapi/operation/common/models.go

```diff
@@ -102,3 +104,74 @@ type RegisteredServicesResponse struct {
 	// in: body
 	Names []string `json:"names"`
 }
+
+// SendNewMessageRequest model
+//
+// This is used for operation to send new message
+//
+// swagger:parameters sendNewMessage
+type SendNewMessageRequest struct {
+	// Params for sending new message
+	//
+	// in: body
+	Params *SendNewMessageParams
+}
+
+// SendNewMessageParams contains parameters for sending new message
+// with three destination options
+//	1. ConnectionID
+//	2. TheirDID
+//	3. ServiceEndpoint (With recipient Keys, endpoint and optional routing keys)
+type SendNewMessageParams struct {
+
+	// Connection ID of the message destination
+	// This parameter takes precedence over all the other destination parameters.
```

Comment:
Can you add more comments detailing when parameters are optional. E.g., ConnectionID is not needed when ...

Reply:
updated